src/macros.rs

```diff
@@ -512,23 +510,21 @@ macro_rules! tarpc_service_then__ {
                   -> ::std::result::Result<$out, $crate::Error<$error>> {
             match msg__ {
                 ::std::result::Result::Ok(msg__) => {
-                    if let Response__::$fn_name(msg__) =
-                        msg__
-                    {
-                        ::std::result::Result::Ok(msg__)
-                    } else {
-                       unreachable!()
+                    #[allow(unreachable_patterns)]
+                    match msg__ {
+                        Response__::$fn_name(msg__) =>
+                            ::std::result::Result::Ok(msg__),
+                        _ => unreachable!(),
```

Comment:
This code is client-side. This code path can be hit by a non-tarpc server implementing the service protocol incorrectly. So right now there's kind of the assumption that the only tarpc protocol-serving servers are running tarpc code.